src/native-state-wayland.cpp

```diff
@@ -121,6 +143,31 @@ NativeStateWayland::registry_handle_global(void *data, struct wl_registry *regis
 
         wl_output_add_listener(my_output->output, &output_listener_, my_output);
         wl_display_roundtrip(that->display_->display);
+    } else if (strcmp(interface, "wl_seat") == 0) {
+        that->display_->seat =
+            static_cast<struct wl_seat *>(
+            wl_registry_bind(registry, id, &wl_seat_interface, 1));
+        wl_seat_add_listener(that->display_->seat, &seat_listener_, that);
+    } else if (strcmp(interface, "wl_shm") == 0) {
+        that->display_->shm =
+            static_cast<struct wl_shm *>(
+                wl_registry_bind(registry, id, &wl_shm_interface, 1));
+
+        struct my_cursor *my_cursor = new struct my_cursor();
```

Comment:
We need to properly free this struct and members, if we don't end up using it (e.g., can't get cursor theme or default cursor below). If manual management complicates the code too much, perhaps it would be worth having a constructor/destructor for struct my_cursor.

Alternatively, we could have my_cursor itself always be a valid part of my_display, with my_cursor.* members being nullptr in case of failure.

Reply:
Update to include proper clean-ups in case of failures/errors.